src/app/screens/auth/signup/complete-data/index.tsx

```diff
@@ -99,6 +106,24 @@ export const CompleteDataScreen = ({
           >
             <ButtonText>{t('common:button.confirm')}</ButtonText>
           </Button>
+
+          <DatePicker
+            modal
+            open={showDatePicker}
+            date={selectedDate || new Date()}
+            mode="date"
+            maximumDate={new Date()}
+            title={t('inputs:placeholder.datepicker')}
+            locale={RNLocalize.getLocales()[0].languageCode}
+            onConfirm={date => {
+              setShowDatePicker(false);
+              setSelectedDate(date);
+              setValue('birthDate', date.toLocaleDateString());
+            }}
+            onCancel={() => {
+              setShowDatePicker(false);
+            }}
+          />
```

Comment:
Mmm I think that you will need to use the <Controller/> from react hook form, so the form state be controlled only by the library

Reply:
we're using setValue from react hook form to map the value to useForm hook  
https://github.com/routine-fit/mobile-app/blob/cb3086da0b98b3c67cc692cfd6faa82fcab9974c/src/app/screens/auth/signup/complete-data/index.tsx#L30

https://github.com/routine-fit/mobile-app/blob/cb3086da0b98b3c67cc692cfd6faa82fcab9974c/src/app/screens/auth/signup/complete-data/index.tsx#L121

Reply:
But why do you use the setSelectedDate for? If you are using the react hook form state

Reply:
according to docs we need that state to handle the date prop on DatePicker component, overall we're using DatePicker component as a mechanism to select the date but then we're using a ControlledInput to handle the validation on the form, the DatePicker modal is not part of the form since the library only provides the modal and we cannot incapsulate it on a form. 
On first instance i tried to create a shared DatePicker componen as we did with controlledInput and ControlledSelectInput, but if i wanted to present a modal as a way to select data that was not feasible, so i decided to use a the Modal provided by the library to select the values and map the value directly to a ControlledInput which will handle the validations  

```ts
 <ControlledInput
            controller={{
              control,
              name: 'birthDate',
            }}
            formControlProps={{
              mb: '$4',
            }}
            inputProps={{
              onTouchStart: () => setShowDatePicker(true),
            }}
          />
```